docs/reference-guide/modules/commands/pages/command-handlers.adoc

```diff
@@ -1,249 +1,926 @@
+Axon automatically loads the entity instance based on the command's routing key:
+
+[source,java]
+----
+import org.axonframework.messaging.commandhandling.annotation.CommandHandler;
+import org.axonframework.extension.spring.stereotype.EventSourced;
+import org.springframework.stereotype.Component;
+
+// The entity definition with state and event sourcing handlers
+@EventSourced
+public class GiftCard {
+
+    private String id;
+    private int remainingValue;
+
+    @EventSourcingHandler
+    private void on(CardIssuedEvent event) {
+        this.id = event.cardId();
+        this.remainingValue = event.amount();
+    }
+
+    @EventSourcingHandler
+    private void on(CardRedeemedEvent event) {
+        this.remainingValue -= event.amount();
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public int getRemainingValue() {
+        return remainingValue;
+    }
+
+    protected GiftCard() {
+    }
+}
+
+// Separate command handlers that receive the entity as a parameter
+@Component // <1>
+public class GiftCardCommandHandlers {
+
+    @CommandHandler // <2>
+    public String handle(IssueCardCommand cmd, EventAppender eventAppender) {
+        // Creational handler - no entity parameter needed
+        GiftCard card = new GiftCard();
+        eventAppender.append(new CardIssuedEvent(cmd.cardId(), cmd.amount()));
+        return cmd.cardId();
+    }
+
+    @CommandHandler // <3>
+    public void handle(RedeemCardCommand cmd, GiftCard card, EventAppender eventAppender) {
+        // Instance handler - entity automatically loaded by Axon
+        if (cmd.amount() <= 0) {
             throw new IllegalArgumentException("amount <= 0");
         }
-        if (cmd.getAmount() > remainingValue) {
+        if (cmd.amount() > card.getRemainingValue()) { // <4>
             throw new IllegalStateException("amount > remaining value");
         }
-        apply(new CardRedeemedEvent(id, cmd.getTransactionId(), cmd.getAmount()));
+        eventAppender.append(new CardRedeemedEvent(card.getId(), cmd.transactionId(), cmd.amount()));
     }
-    // omitted event sourcing handlers
 }
+----
+
+<1> Command handlers are defined in a separate component (not on the entity itself).
+
+<2> Creational handlers don't receive an entity parameter—they create the entity.
+
+<3> Instance handlers receive the entity as a parameter.
+Axon automatically loads it based on the command's routing key.
+
+<4> The handler accesses entity state through getter methods to make decisions.
+
+[TIP]
+====
+The command-centric approach is useful when:
+
+* You want to keep command handling logic separate from entity state management
+* You have multiple entities that share similar command handling patterns
+* You prefer to organize code by command/use case rather than by entity
+* You want to easily test command handling logic without the entity class
```

Comment:
Shouldn't this tip mention slice-based architecture as one of the reasons to go for this approach?